lib/src/views/Calendar/Calendar.tsx

```diff
@@ -356,9 +71,152 @@ export const styles = (theme: Theme) => ({
     display: 'flex',
     justifyContent: 'center',
   },
-});
+  iconButton: {
+    zIndex: 1,
+    backgroundColor: theme.palette.background.paper,
+  },
+  previousMonthButton: {
+    marginRight: 12,
+  },
+  daysHeader: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  dayLabel: {
+    width: 36,
+    height: 40,
+    margin: '0 2px',
+    textAlign: 'center',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    color: theme.palette.text.hint,
+  },
+}));
+
+export const Calendar2: React.FC<CalendarProps> = ({
+  date,
+  onChange,
+  minDate,
+  maxDate,
+  slideDirection,
+  disableFuture,
+  disablePast,
+  currentMonth,
+  renderDay,
+  allowKeyboardControl,
+  ...props
+}) => {
+  const utils = useUtils();
+  const theme = useTheme();
+  const classes = useStyles();
+  const now = utils.date();
+  const variant = React.useContext(VariantContext);
+
+  const validateMinMaxDate = React.useCallback(
+    (day: MaterialUiPickersDate) => {
+      return Boolean(
+        (disableFuture && utils.isAfterDay(day, now)) ||
+          (disablePast && utils.isBeforeDay(day, now)) ||
+          (minDate && utils.isBeforeDay(day, utils.date(minDate))) ||
+          (maxDate && utils.isAfterDay(day, utils.date(maxDate)))
+      );
+    },
+    [disableFuture, disablePast, maxDate, minDate, now, utils]
+  );
+
+  const shouldDisableDate = React.useCallback(
+    (day: MaterialUiPickersDate) => {
+      return (
+        validateMinMaxDate(day) || Boolean(props.shouldDisableDate && props.shouldDisableDate(day))
+      );
+    },
+    [props, validateMinMaxDate]
+  );
+
+  const handleDaySelect = React.useCallback(
+    (day: MaterialUiPickersDate, isFinish = true) => {
+      onChange(utils.mergeDateAndTime(day, date), isFinish);
+    },
+    [date, onChange, utils]
+  );
+
+  const moveToDay = React.useCallback(
+    (day: MaterialUiPickersDate) => {
+      if (day && !shouldDisableDate(day)) {
+        handleDaySelect(day, false);
+      }
+    },
+    [handleDaySelect, shouldDisableDate]
+  );
+
+  useKeyDown(Boolean(allowKeyboardControl && variant !== 'static'), {
```

Comment:
There is some issues with this useKeyDown helper:
- It doesn't use the synthetic event, so he can't use `event.key` (React polyfills it)
- It doesn't listen to the active frame, I believe it wouldn't work when rendered in an iframe, popup